test/01_typedef.jl

```diff
@@ -37,15 +41,17 @@ module TestTypeDef
     @test pool.order[3] === DefaultRefType(3)
 
     for i in 1:3
-        x = CategoricalValue(i, pool)
+        x = catvalue(i, pool)
 
-        @test isa(x, CategoricalValue)
+        @test iscatvalue(x)
+        @test valtype(typeof(x)) == String
+        @test reftype(typeof(x)) == DefaultRefType
```

Comment:
Please also test directly `x isa CategoricalValue` or even  `x isa CategoricalValue{Int, DefaultRefType}`, that doesn't hurt. Same below.